src/utils/actions/writeAction.ts

```diff
@@ -1,24 +1,65 @@
-import { mutate } from '~amplifyClient';
-import { ContractEvent } from '~types';
+import { Extension, getExtensionHash } from '@colony/colony-js';
+import { mutate, query } from '~amplifyClient';
+import { ColonyActionType, ContractEvent } from '~types';
 import { verbose } from '~utils';
 
+interface ActionFields extends Record<string, any> {
+  initiatorAddress: string;
+  type: ColonyActionType;
+}
+
 export const writeActionFromEvent = async (
   event: ContractEvent,
   colonyAddress: string,
-  actionFields: Record<string, any>,
+  actionFields: ActionFields,
 ): Promise<void> => {
   const { transactionHash, blockNumber, timestamp } = event;
 
   const actionType = actionFields.type ?? 'UNKNOWN';
-  verbose('Action', actionType, 'took place in Colony:', colonyAddress);
+  const showInActionsList = await showActionInActionsList(
+    colonyAddress,
+    actionFields.initiatorAddress,
+  );
 
+  verbose('Action', actionType, 'took place in Colony:', colonyAddress);
   await mutate('createColonyAction', {
     input: {
       id: transactionHash,
       colonyId: colonyAddress,
       blockNumber,
       createdAt: new Date(timestamp * 1000).toISOString(),
+      showInActionsList,
       ...actionFields,
     },
   });
 };
+
+const getVotingReputationInstallations = async (
+  colonyAddress: string,
+): Promise<Array<{ id: string }> | undefined> => {
+  const votingRepHash = getExtensionHash(Extension.VotingReputation);
+  const { items } =
+    (await query<{ items: Array<{ id: string }> }>(
+      'getVotingRepInstallations',
+      {
+        votingRepHash,
+        colonyAddress,
+      },
+    )) ?? {};
+
+  return items;
+};
+
+const showActionInActionsList = async (
```

Comment:
We put an action in the actions list if it wasn't created by a motion. This is to avoid showing a duplicate once the motion has been finalized.

Reply:
Could we potentially check for that when the action is handled instead? And not write it to the DB at all? I don't know if we need that info, but it'd simplify all the surrounding logic